### ISSUE TITLE
Optimize FilteredPlayers()

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#1350]
Calculate it once and store the result instead of recalculating all the time.

Also fix the number of players shown when friends only is selected as filter. It should still show the total number of players on the server, not the number of friends on the server.

@timakro Are you working on this?

**Comment by timakro** [comment]
I ran into problems modifying `CServerInfo` objects which are declared as constant in most of the serverstatus code and gave up. There are in general problems with the serverbrowser performance since the spoofing attacks because of the many fake players the client has to handle. If somebody wants to tackle this I noticed that two changes independently reduce the serverbrowser lag significantly (and combined almost fix it for me):
1. Going back to the old less expensive way player numbers per server were calculated (or hopefully implementing the new method in a more efficient way)
2. Checking at the top of `CFriends::GetFriendState` if the name is `"(connected)"` and in that case assuming it's not a friend.
